### ISSUE TITLE
Hangs and does not work on mac

### ISSUE THREAD

**Issue by raine** [wilsonzlin/minify-html#6]
```sh
% curl https://wilsonl.in/minify-html/bin/0.3.3-macos-x86_64 > minify-html
% chmod +x minify-html
% curl -O https://www.bike24.com/p2388156.html
% ./minify-html --src p2388156.html --out temp.html
[nothing happens]
```

**Comment by wilsonzlin** [comment]
Thanks for raising, there was a small bug when creating the source file representation for error messages, which I just fixed and released in version `0.3.4`. Allow a few minutes for the build to complete.

**Comment by raine** [comment]
Thanks. I minified one page successfully with the new version, but promptly found another one that hangs.

```
% curl -O https://www.wiggle.co.uk/muc-off-dry-shower-100ml
% minify-html --src muc-off-dry-shower-100ml --out temp.html
```

**Comment by raine** [comment]
Nevermind, I thought I had the latest version but didn't. 

It does not hang, but just fails to some error with the html.

Before when I tried [`html-minifier`](https://github.com/kangax/html-minifier) on these same files, I did not get any parse errors. Is your tool just strict about structure of the input HTML? I wonder if it's a non-starter to try to use it with 3rd party HTML that you can't have any guarantees about.

**Comment by wilsonzlin** [comment]
Can you provide the source HTML/URLs and any error messages for your other files? I do want to figure out and fix any parsing issues, but it's impossible to do so without any context or reference. Examples are best for helping improve minify-html, so the more examples the better :+1: 

minify-html is not very strict, and there are only three possible errors in minify-html, defined in [err.rs](https://github.com/wilsonzlin/minify-html/blob/master/src/err.rs). The page you originally referred to did not work due to tag omission e.g. `<li>` without `</li>`. This is allowed by the spec but was difficult to implement and originally disallowed, as mentioned in the README of previous versions. However, I've just implemented support for it in version `0.3.5`, and the page does work now; feel free to try it out on your other pages and let me know how they go.

I'm not sure what you mean by "3rd party HTML" and no "guarantees", as HTML is a spec. I aim to parse most HTML code and will happily look into any that don't.

**Comment by wilsonzlin** [comment]
Closing due to inactivity, feel free to comment if issue persists.
